src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  onClose?: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo: { title, id, user, completed },
+  onClose = () => {},
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(user?.id || 0).then(() => setIsLoading(false));
```

Comment:
You're fetching data inside your component, which is a side effect. In React, side effects should be encapsulated in `useEffect`. However, you're not handling loading and error states properly. When `getUser` is pending, you set `isLoading` to true, but you never set it back to false if the promise rejects. This can lead to an infinite loading state. Consider adding error handling to your `useEffect` to address this issue.